PyRDF/backend/Dist.py

```diff
@@ -209,7 +211,8 @@ def _get_clustered_ranges(self, nentries, npartitions, treename, filelist):
             offset_first_cluster = clusters[index_start][2]
             ranges.append(Range(start - offset_first_cluster,
                                 end - offset_first_cluster,
-                                range_files))
+                                range_files,
+                                friend_filelist))
```

Comment:
`range_friend_files`

Reply:
I am in doubt about this one particularly. Thing is that `range_files` is created after retrieving the file name for that particular range of entries, but that doesn't automatically mean that for the same range of entries we would have the same friend file names, right?

Reply:
Right, this might require further investigation (also to see how it is currently implemented in root), as clusters in both sets of files may not be the same. 

`range_files` is added to the list of ranges for debugging purposes and to create `TChain`'s with a reduced size so we do not add unused files. At some point we would need to do the same for friend trees/chains but for now we can focus on the creation of the friend chain and add this optimization later on.